drizzlepac/haputils/poller_utils.py

```diff
@@ -491,16 +491,19 @@ def parse_mvm_tree(det_tree, all_mvm_exposures, log_level):
                 prod_info = (filename[0] + " " + filetype).lower()
                 #
                 # mvm prod_info = 'skycell_p1234_x01y01 wfc3 uvis f200lp all 2009 1 drz'
+                
+                # prod_list is 0: proposal_id, 1:observation_id, 2:instrument, 3:detector, 
+                # 4:aperture_from_poller, 5:filename, 6:filters, 7:filetype
                 #
                 prod_list = prod_info.split(" ")
                 multi_scale = prod_list[2].upper() in ['IR', 'PC']
                 pscale = 'fine' if not multi_scale else 'coarse'
                 prod_info += " {:s}".format(pscale)
 
                 if prod_list[5].strip() != '':
-                    layer = (prod_list[3], pscale, prod_list[4], prod_list[5])
+                    layer = (prod_list[3], pscale, prod_list[5], prod_list[6])
                 else:
-                    layer = (prod_list[3], pscale, prod_list[4])
+                    layer = (prod_list[3], pscale, prod_list[5])
 
                 ftype = prod_list[-1]
                 cellid = prod_list[0].split('-')[1]
```

Comment:
Add a comment before prop_id that prop_id is really the projection cell number, and obset_id represents the X and Y skycell coordinates.  I understand now why this code confused you.  The terminology is SVM and the SkyCell* classes should have had a different constructor signature!